From 929925ca5d4541bffcc595cbaedafed32dde6558 Mon Sep 17 00:00:00 2001
From: Lee Marlow <lee.marlow@gmail.com>
Date: Thu, 17 Jul 2008 16:07:18 -0600
Subject: [PATCH] inline ActiveSupport's String#starts_with? implementation since it may not be loaded yet

allows starting server with mongrel_rails, etc., instead of just script/server
---
 railties/lib/rails/gem_dependency.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/gem_dependency.rb b/railties/lib/rails/gem_dependency.rb
index a1b6d52..394c8d3 100644
--- a/railties/lib/rails/gem_dependency.rb
+++ b/railties/lib/rails/gem_dependency.rb
@@ -126,7 +126,7 @@ module Rails
 
       def find_frozen_gem
         return nil if @requirement.nil?
-        a = Dir.entries(self.class.unpacked_path).select{|d| d.starts_with? @name and File.directory? File.join(self.class.unpacked_path, d) and @requirement.satisfied_by? Gem::Version.create(d.split('-').last)}
+        a = Dir.entries(self.class.unpacked_path).select{|d| d[0, @name.length] == @name and File.directory? File.join(self.class.unpacked_path, d) and @requirement.satisfied_by? Gem::Version.create(d.split('-').last)}
         a.sort{|x,y| x.split('-').last <=> y.split('-').last}.last
       end
   end
-- 
1.5.6.3.439.g1e10.dirty

